packages/beacon-node/src/chain/bls/utils.ts

```diff
@@ -29,3 +30,57 @@ export function getAggregatedPubkeysCount(signatureSets: ISignatureSet[]): numbe
   }
   return pubkeysConut;
 }
+
+/**
+ * Splits an array into an array of arrays maximizing the size of the smallest chunk.
+ */
+export function chunkifyMaximizeChunkSize<T>(arr: T[], minPerChunk: number): T[][] {
+  const chunkCount = Math.floor(arr.length / minPerChunk);
+  if (chunkCount <= 1) {
+    return [arr];
+  }
+
+  // Prefer less chunks of bigger size
+  const perChunk = Math.ceil(arr.length / chunkCount);
+  const arrArr: T[][] = [];
+
+  for (let i = 0; i < arr.length; i += perChunk) {
+    arrArr.push(arr.slice(i, i + perChunk));
+  }
+
+  return arrArr;
+}
+
+/**
+ * `rand` must not be exactly zero. Otherwise it would allow the verification of invalid signatures
+ * See https://github.com/ChainSafe/blst-ts/issues/45
+ */
+export function randomBytesNonZero(bytesCount: number): Uint8Array {
+  const rand = crypto.getRandomValues(new Uint8Array(bytesCount));
+  for (let i = 0; i < bytesCount; i++) {
+    if (rand[i] !== 0) return rand;
+  }
+  rand[0] = 1;
+  return rand;
+}
+
+/**
+ * Fully deserializes a signature set that may have a serialized signature
+ * NOTE: This can throw for invalid signature and should be wrapped in a try/catch
+ */
+export function deserializeSet(set: WorkRequestSet): SignatureSet {
+  return {
+    message: set.message,
+    publicKey: set.publicKey,
+    signature:
+      set.signature instanceof bls.Signature
+        ? set.signature
+        : bls.Signature.fromBytes(set.signature, CoordType.affine, true),
```

Comment:
no need to verify signature for same message verification because we verified it alreayd